### PR TITLE
[v7r3] added a flag for disabling the use of SecurityLogging service

### PR DESCRIPTION
**Pull request by fstagni**
See https://gitlab.cern.ch/ai/it-puppet-module-dirac/-/merge_requests/75

BEGINRELEASENOTES

*Framework
CHANGE: added a flag for disabling the use of SecurityLogging service

ENDRELEASENOTES

### CODE REVIEW

**Comment by fstagni** [comment]
Seems like there's a `psutil` bug: https://github.com/giampaolo/psutil/issues/2048

**Comment by fstagni** [comment]
https://github.com/DIRACGrid/DIRAC/pull/5760/commits/54b6db3ca86556d8d119c8f6aab48a04454ea469 is very much subject to review from @chaen @TaykYoku

File: src/DIRAC/Core/Tornado/Server/TornadoService.py [hunk 1]
```diff
@@ -121,6 +123,8 @@ def export_streamToClient(self, myDataToSend, token):
     # We also need to add specific attributes for each service
     _monitor = None
 
+    SVC_SECLOG_CLIENT = SecurityLogClient()
```
Comment (TaykYoku) [inline src/DIRAC/Core/Tornado/Server/TornadoService.py:126]: If I understand correctly, it works similarly to `MonitoringClient`, so `SecurityLogClient` can be placed in the same way: initialisation in the `_initMonitoring`.

**Review by TaykYoku — COMMENTED** [review]
`SecurityLogClient` uses `DIRAC.Core.Utilities.ThreadScheduler` for reporting.
Since we are moving from **DISET** to **Tornado**, it is worth using `tornado.web.IOLoop` to run periodic tasks. I think the [following PR](https://github.com/DIRACGrid/DIRAC/pull/5771) will be useful for this implementation.

**Comment by chaen** [comment]
Given that we want to ditch it, I really do not think it is worth the pain of adding the `securityLog` to `TornadoService`

**Comment by fstagni** [comment]
> Given that we want to ditch it, I really do not think it is worth the pain of adding the `securityLog` to `TornadoService`

Yes, we want to ditch the `securityLog`, but ATM it's still needed. In order to ditch it, we need in fact to store on ES 6 months worth of log data, for each of the running DIRAC services. This is not impossible but rather expensive, considering that ATM we are correctly sending to ES everything that we put in `verbose` level. What is stored on the current `securityLog` is a rather minimal subset of that. So, ditching "tomorrow" the current `securityLog` service is not possible. 

What we can do is to use again ES, in combination with logstash: meaning that we create ES indices with the content of what would normally go in the CSV files created by `SecurityLog` services, driven by a logstash rule. Would this work?

**Comment by chaen** [comment]
yes it would work. But I still would not add it to `Tornado` now

File: src/DIRAC/Core/DISET/private/Service.py [hunk 2]
```diff
@@ -413,7 +416,8 @@ def _processInThread(self, clientTransport):
             if monReport:
                 self.__endReportToMonitoring(*monReport)
 
-    def _createIdentityString(self, credDict, clientTransport=None):
+    @staticmethod
+    def _createIdentityString(credDict, clientTransport=None):
```
[the remark this answers is not in the document]
Comment (fstagni) [reply src/DIRAC/Core/DISET/private/Service.py:420]: Why? it's just a static method

**Comment by fstagni** [comment]
I have removed the changes to tornado, and added changes for logstash configuration. Please review this PR, and the one for puppet/logstash. If you are happy we can try this in the coming hackathon.

File: src/DIRAC/Core/DISET/private/Service.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (TaykYoku) [reply src/DIRAC/Core/DISET/private/Service.py:420]: sorry, of course you're right!

File: docs/source/AdministratorGuide/Configuration/ConfReference/Systems/Framework/Services/index.rst [hunk 3]
```diff
@@ -28,6 +28,9 @@ presented in the following table:
 +---------------------------------------------------------------------------+--------------------------------+
 | *Authorization/Default*    | Define to who is required the authorization  | Default = all                  |
 +---------------------------------------------------------------------------+--------------------------------+
+| *EnableSecurityLogging*    | This flag is used to disable the use of      | EnableSecurityLogging = False  |
```
Comment (chaen) [inline docs/source/AdministratorGuide/Configuration/ConfReference/Systems/Framework/Services/index.rst:31]: this doc is going away, it should rather go in the `configTemplate.cfg`

File: docs/source/AdministratorGuide/Configuration/ConfReference/Systems/Framework/Services/SecurityLogging/index.rst [hunk 4]
```diff
@@ -1,7 +1,8 @@
 Systems / Framework / <INSTANCE> / Service / SecurityLogging - Sub-subsection
 =============================================================================
 
-SecurityLogging service is used by all server to log all connections.
+SecurityLogging service can be used by all services to log all connections, for security-related purpose.
```
Comment (chaen) [inline docs/source/AdministratorGuide/Configuration/ConfReference/Systems/Framework/Services/SecurityLogging/index.rst:4]: same here, the `ConfReference` is going away

**Comment by chaen** [comment]
The code and puppet code seems fine. Just you need to add the option doc to the `dirac.cfg` instead of the `ConfReference`

**Review by chaen — DISMISSED** [review]

**Pull request merged**